src/java.base/share/classes/java/math/BigDecimal.java

```diff
@@ -1372,10 +1372,15 @@ static BigDecimal zeroValueOf(int scale) {
      * by the {@link Double#toString(double)} method.
      *
      * @apiNote This is generally the preferred way to convert a
-     * {@code double} (or {@code float}) into a {@code BigDecimal}, as
+     * {@code double} into a {@code BigDecimal}, as
      * the value returned is equal to that resulting from constructing
      * a {@code BigDecimal} from the result of using {@link
      * Double#toString(double)}.
+     * <p>
+     * While a {@code float} argument {@code v} can be passed to this method,
+     * the result usually contains too many trailing digits compared
+     * to the precision of a {@code float}.
+     * Consider using {@code new BigDecimal(Float.toString(v))} instead.
```

Comment:
Perhaps move "Consider using" to the previous line; otherwise, looks fine.

Reply:
I usually start a sentence on a new line because that generates less noise when diffing in the future.
The HTML renders it in the same paragraph as the preceding text.